src/Nethermind/Nethermind.Optimism/OptimismPlugin.cs

```diff
@@ -0,0 +1,326 @@
+
+        _api.BlockProducerEnvFactory = new OptimismBlockProducerEnvFactory(
+            _api.ChainSpec,
+            _api.DbProvider,
+            _api.BlockTree,
+            _api.ReadOnlyTrieStore,
+            _api.SpecProvider,
+            _api.BlockValidator,
+            _api.RewardCalculatorSource,
+            _api.ReceiptStorage,
+            _api.BlockPreprocessor,
+            _api.TxPool,
+            _api.TransactionComparerProvider,
+            _blocksConfig,
+            _api.LogManager);
+
+        _api.GasLimitCalculator = new OptimismGasLimitCalculator();
+
+        _blockProductionTrigger = new BuildBlocksWhenRequested();
+
+        BlockProducerEnv producerEnv = _api.BlockProducerEnvFactory.Create();
+
+        _api.BlockProducer = _blockProducer = new OptimismPostMergeBlockProducer(
+            new OptimismPayloadTxSource(),
+            producerEnv.TxSource,
+            producerEnv.ChainProcessor,
+            producerEnv.BlockTree,
+            _blockProductionTrigger,
+            producerEnv.ReadOnlyStateProvider,
+            _api.GasLimitCalculator,
+            NullSealEngine.Instance,
+            new ManualTimestamper(),
+            _api.SpecProvider,
+            _api.LogManager,
+            _api.Config<IBlocksConfig>());
+
+        return Task.FromResult(_api.BlockProducer);
+    }
+
+    #endregion
+
+    public INethermindApi CreateApi(IConfigProvider configProvider, IJsonSerializer jsonSerializer,
+        ILogManager logManager, ChainSpec chainSpec) =>
+        new OptimismNethermindApi(configProvider, jsonSerializer, logManager, chainSpec);
+
+    public Task Init(INethermindApi api)
+    {
+        if (!ShouldRunSteps(api))
+            return Task.CompletedTask;
+
+        _api = (OptimismNethermindApi)api;
+        _mergeConfig = _api.Config<IMergeConfig>();
+        _syncConfig = _api.Config<ISyncConfig>();
+        _blocksConfig = _api.Config<IBlocksConfig>();
+        _logger = _api.LogManager.GetClassLogger();
+
+        ArgumentNullException.ThrowIfNull(_api.BlockTree);
+
+        _api.PoSSwitcher = AlwaysPoS.Instance;
+
+        _blockCacheService = new BlockCacheService();
+        _api.InvalidChainTracker = _invalidChainTracker = new InvalidChainTracker(
+            _api.PoSSwitcher,
+            _api.BlockTree,
+            _blockCacheService,
+            _api.LogManager);
+        _api.DisposeStack.Push(_invalidChainTracker);
+
+        _api.FinalizationManager = _blockFinalizationManager = new ManualBlockFinalizationManager();
+
+        _api.RewardCalculatorSource = NoBlockRewards.Instance;
+        _api.SealValidator = NullSealEngine.Instance;
+        _api.GossipPolicy = ShouldNotGossip.Instance;
+
+        _api.BlockPreprocessor.AddFirst(new MergeProcessingRecoveryStep(_api.PoSSwitcher));
+
+        return Task.CompletedTask;
+    }
+
+    public Task InitNetworkProtocol() => Task.CompletedTask;
+
+    public Task InitSynchronization()
```

Comment:
Why this is needed?

Reply:
WHy do we have to init synchronization?

Reply:
we would initialize syncronization anyways in the default `InitializeNetwork`, at least this one would work if p2p was enabled in Optimism.